journal.py

```diff
@@ -139,7 +143,13 @@ def add_entry(request):
         except psycopg2.Error:
             # This will catch any errors generated by the database
             return HTTPInternalServerError
-        return HTTPFound(request.route_url('home'))
+        cursor = request.db.cursor()
+        cursor.execute(NEWEST_ENTRY)
+        keys = ('id', 'title', 'text', 'created')
+        entry = dict(zip(keys, cursor.fetchone()))
+        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
+        entry['created'] = entry['created'].strftime('%b. %d, %Y')
```

Comment:
This block of code gets repeated in a number of places with only minor variations?  Could it be refactored into a helper method?